src/lib/components/workflow/WorkflowTaskSelection.svelte

```diff
@@ -36,17 +56,49 @@
 		});
 		selectionTasksNames = Array.from(selectionTasks.keys());
 
+		setSelectionControlData();
 	}
 
-	function setSelectedGroup(group) {
-		console.log(group);
-		selectedTypeOfTask = group;
+	function setSelectionControlData() {
+		let optionsMap = [];
+
+		if (selectedTypeOfTask === 'common') {
+			// Group filtered tasks by source
+			optionsMap = filteredTasks.reduce((dataOptions, task) => {
+				const source = task.source.split(':')[1]; // Package name
```

Comment:
To get the package a task belongs to, I split its source property and take the second (1) element of the split string.
Is it sound enough for every case? @tcompa

Reply:
Could it happen that a task has no source or edge cases to consider?

Reply:
I'll let Tommaso analyse this in detail. My understanding is that common tasks will always have a very predictable source structure, but user tasks could have any string as source, as long as it's unique.

Reply:
This is all sound.

By construction, our common tasks come from the installation&collection of a Python package, and the source looks like `pip_remote:fractal_tasks_core:0.9.0:::cellpose_segmentation` or `pip_remote:fractal_tasks_core:0.10.0:fractal-tasks::copy_ome-zarr_structure`, where the second field is indeed the package name.

Non-common tasks, on the contrary, only have a single constraint on the source, which is that the first field should match with `task.owner`.

These structures for the task sources are not strictly enforced, but they will always be there "by construction". Therefore they will hold until we make significant changes to the API (e.g. if we wanted to allow a change of owner for a given task, then it wouldn't be obvious what the source should be).

What is guaranteed is that all tasks have a source, and that this source has to be unique.

TL;DR
@rkpasia you are relying on very reasonable assumptions. They could evolve in the future, but that would only come with new use cases which we are currently not supporting.